src/Microsoft.Azure.Functions.Extensions.Mcp/Http/StreamableHttpRequestHandler.cs

```diff
@@ -2,18 +2,214 @@
 // Licensed under the MIT License.
 
 using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.Functions.Extensions.Mcp.Configuration;
+using Microsoft.Azure.Functions.Extensions.Mcp.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using ModelContextProtocol;
 using ModelContextProtocol.Protocol;
+using ModelContextProtocol.Server;
+using System.IO.Pipelines;
 using System.Text.Json.Serialization.Metadata;
+using static Microsoft.Azure.Functions.Extensions.Mcp.McpConstants;
 
 namespace Microsoft.Azure.Functions.Extensions.Mcp;
 
-internal sealed class StreamableHttpRequestHandler : IStreamableHttpRequestHandler
+internal sealed class StreamableHttpRequestHandler(
+    IMcpInstanceIdProvider instanceIdProvider,
+    IMcpClientSessionManager clientSessionManager,
+    IServiceProvider applicationServices,
+    IOptions<McpServerOptions> mcpServerOptions,
+    IOptions<McpOptions> mcpOptions,
+    ILoggerFactory loggerFactory) : IStreamableHttpRequestHandler
 {
-    private static readonly JsonTypeInfo<JsonRpcError> ErrorTypeInfo = (JsonTypeInfo<JsonRpcError>)McpJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonRpcError));
+    private static readonly JsonTypeInfo<JsonRpcError> ErrorTypeInfo =
+        (JsonTypeInfo<JsonRpcError>) McpJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonRpcError));
 
     public Task HandleRequest(HttpContext context)
     {
-        throw new NotImplementedException();
+        if (HttpMethods.IsPost(context.Request.Method))
+        {
+            return HandlePostRequest(context);
+        }
+
+        context.Response.StatusCode = StatusCodes.Status405MethodNotAllowed;
+        return Task.CompletedTask;
+    }
+
+    private async Task HandlePostRequest(HttpContext context)
+    {
+        if (!await PostHasValidMediaTypesAsync(context))
+        {
+            return;
+        }
+
+        var session = await GetOrCreateSessionAsync(context, mcpOptions.Value);
+
+        if (session is null)
+        {
+            return;
+        }
+
+        try
+        {
+            McpHttpUtility.SetSseContext(context);
+
+            var responseWritten = await session.Transport.HandlePostRequest(new HttpDuplexPipe(context), context.RequestAborted);
+
+            if (!responseWritten)
+            {
+                context.Response.Headers.ContentType = (string?)null;
+                context.Response.StatusCode = StatusCodes.Status202Accepted;
+            }
+        }
+        finally
+        {
+            if (session.Transport.IsStateless)
```

Comment:
While all of the logic in the current set of changes is dealing with stateless sessions only, some logic to prepare for stateful sessions is already in place (that work is in progress).